macros/src/main/scala/com/avast/cactus/CactusMacros.scala

```diff
@@ -102,27 +97,31 @@ object CactusMacros {
 
       tree
     }
+
+
+    c.Expr[Converter[GpbClass, CaseClass]] {
+      q"""
+         new Converter[$gpbType, $caseClassType] {
```

Comment:
It's really hard for me to follow the code so take this as a guess. Have you thought about how many instances of `Converter` you make? Could you make just one for each type so that it's optimized? Maybe I am completely wrong.

Reply:
It was not possible until now (well, unless you created some methods which took care about the conversion and used Cactus inside), however now you can easily "save" created converters on a single place to optimize it. So you're not wrong :-)
What was lost is reporting your variable name in the failure, I will try to think about it but I'm afraid it's not possible anymore.

Reply:
Maybe you should have attended the macros workshop to be able to read the code better :-P